op-program/chainconfig/testing.go

```diff
@@ -0,0 +1,76 @@
+package chainconfig
+
+import (
+	"encoding/json"
+	"fmt"
+	"math"
+	"os"
+
+	"github.com/ethereum-optimism/optimism/op-node/rollup"
+	"github.com/ethereum/go-ethereum/params"
+)
+
+var (
+	TestChainID = uint64(math.MaxUint64)
+
+	testRollupConfigPath = "fpp_rollup_config.json"
+	testL2GenesisPath    = "fpp_l2_genesis.json"
```

Comment:
Using fixed file paths is likely to cause conflicts between different tests.  Couple of options I can see:

1. Create a temp dir per test and write these files to the test specific temp dir (using these fixed names).  Then when executing `op-program` make sure its working directory is set to the temp dir.  The client program can then just read the hard coded file name and it will get the right file for that test.
2. Rather than reading from a file, if the chain ID is unknown, request the JSON network config from the pre-image oracle using another local key (ie fallback to the way we're doing it today).  On-chain only the chain ID would work and so maybe we require an additional flag to enable serving these JSON versions of the chain config or something like that.

Reply:
went with option 2. The diff turned out to be smaller than I expected.